src/main/java/grabber/sources/wnmtl_org.java

```diff
@@ -0,0 +1,112 @@
+package grabber.sources;
+
+import grabber.Chapter;
+import grabber.GrabberUtils;
+import grabber.Novel;
+import grabber.NovelMetadata;
+
+import org.jsoup.HttpStatusException;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class wnmtl_org implements Source{
+    private final String name = "wnmtl.org";
+    private final String url = "https://wnmtl.org";
+    private final boolean canHeadless = false;
+    private Novel novel;
+    private Document toc;
+
+    public wnmtl_org(Novel novel) {
+        this.novel = novel;
+    }
+    public wnmtl_org() {}
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getUrl() {
+        return url;
+    }
+
+    @Override
+    public boolean canHeadless() {
+        return canHeadless;
+    }
+
+    @Override
+    public List<Chapter> getChapterList() {
+        List<Chapter> cList = new ArrayList<>();
+        try {
+            toc = Jsoup.connect(novel.novelLink)
+                    .cookies(novel.cookies)
+                    .userAgent("Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0")
+                    .get();
+            Elements chaps = toc.select("li.wp-manga-chapter > a");
```

Comment:
Chapters are not static and are loaded later  via ajax (POST to admin-ajax.php) see daonovel_com for example

Reply:
> Chapters are not static and are loaded later via ajax (POST to admin-ajax.php) see daonovel_com for example

Hey, I'm trying to learn more about the entire process! Can you tell me how you managed to figure out the correct way to grab the novel? Sorry if I'm being annoying😅

Reply:
If you open the Network tab of the Web Developer tools you can see the http requests the browser makes. 
![img](https://i.imgur.com/BIYSdma.png)

As you can see there are no chapters being loaded on first request. Novel-Grabber uses Jsoup to fetch the sites which is static, meaning it cant work with/execute JavaScript (which loads the chapters later on in the browser).

Now you have to look through the requests to find where the chapters are being loaded, in this case there is a POST to admin-ajax.php

![img](https://i.imgur.com/iWbxxz9.png)

Next you have to  look at what is being POSTed to the server to replicate it.

![img](https://i.imgur.com/DsSlg25.png)

So to make it work in Novel-Grabber you need to first get the toc page, like normal. The toc object is used to fetch the metadate later on and you need to get the novel id for the ajax request. Then you need to fetch the chapter list via a POST. Jsoup can also do this.

![img](https://i.imgur.com/Q67kAhI.png)

Reply:
Thank you so much! Working on fixes now!